server/src/main/java/org/elasticsearch/index/engine/NoOpEngine.java

```diff
@@ -86,4 +102,18 @@ public CacheHelper getReaderCacheHelper() {
             }
         };
     }
+
+    @Override
+    public SegmentsStats segmentsStats(boolean includeSegmentFileSizes, boolean includeUnloadedSegments) {
+        if (includeUnloadedSegments) {
+            final SegmentsStats stats = new SegmentsStats();
+            stats.add(this.stats);
+            if (includeSegmentFileSizes == false) {
+                stats.clearFileSizes();
+            }
+            return stats;
+        } else {
+            return super.segmentsStats(includeSegmentFileSizes, includeUnloadedSegments);
```

Comment:
calling this will fail on a NoopEngine?
Should we instead return an empty stats object?

Reply:
hmm there is a test for this [here](https://github.com/elastic/elasticsearch/pull/39698/files#diff-3fed3ae02b32130e492455f35d2fb3c7R151) that shows that it's not failing but returning an empty object.